packages/presets/client/src/index.ts

```diff
@@ -114,6 +114,10 @@ export const preset: Types.OutputPreset<ClientPresetConfig> = {
       );
     }
     const isPersistedOperations = !!options.presetConfig?.persistedDocuments;
+    if (options.config.semanticNonNull?.errorHandlingClient) {
+      options.schemaAst = await semanticToStrict(options.schemaAst!);
+      options.schema = options.schemaAst as any; // FIXME: `schemaAst` _seems_ to be able to used as `schema`, but not sure if it has any side effects?
```

Comment:
I'm a bit unsure about this:
- `schema` is a `DocumentNode`
- `shemaAst` is a `GraphQLSchema`
- We can forcefully override `schema` with `schemaAst` ... and it works 🤔 

Maybe when loading `schema` in each `generates` block, we do a check? That's my best guess though, and if anyone has more context, please let me know!

Reply:
Maybe what you want here is this?

```suggestion
      options.schema = parse(printSchema(options.schemaAst));
```

Just guessing from the types you mention, I'm not familiar with this codebase.